<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 (Advanced) Working with SQL databases | Bioconductor Annotation Resources</title>
  <meta name="description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 (Advanced) Working with SQL databases | Bioconductor Annotation Resources" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 (Advanced) Working with SQL databases | Bioconductor Annotation Resources" />
  
  <meta name="twitter:description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  

<meta name="author" content="Martin Morgan" />


<meta name="date" content="2020-09-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gene-sets.html"/>

<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor Annotation Resources</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#plan"><i class="fa fa-check"></i><b>1.1</b> Plan</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#bioconductor"><i class="fa fa-check"></i><b>1.2</b> <em>Bioconductor</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html"><i class="fa fa-check"></i><b>2</b> Bulk RNA Sequencing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#rna-seq-differential-expression-data-in-r-bioconductor"><i class="fa fa-check"></i><b>2.1</b> RNA-seq differential expression data in <em>R</em> / <em>Bioconductor</em></a></li>
<li class="chapter" data-level="2.2" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#a-quick-analysis"><i class="fa fa-check"></i><b>2.2</b> A quick analysis</a></li>
<li class="chapter" data-level="2.3" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>2.3</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html"><i class="fa fa-check"></i><b>3</b> Gene symbol mapping</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>3.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="3.2" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#finding-annotation-resources"><i class="fa fa-check"></i><b>3.2</b> Finding annotation resources</a></li>
<li class="chapter" data-level="3.3" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#mapping-between-identiifiers"><i class="fa fa-check"></i><b>3.3</b> Mapping between identiifiers</a></li>
<li class="chapter" data-level="3.4" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>3.4</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html"><i class="fa fa-check"></i><b>4</b> Discovering genomic coordinates</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>4.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="4.2" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#genomic-ranges"><i class="fa fa-check"></i><b>4.2</b> Genomic ranges</a></li>
<li class="chapter" data-level="4.3" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#mapping-from-identifiers-to-ranges"><i class="fa fa-check"></i><b>4.3</b> Mapping from identifiers to ranges</a></li>
<li class="chapter" data-level="4.4" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#flexible-queries-on-ensdb-objects"><i class="fa fa-check"></i><b>4.4</b> Flexible queries on EnsDb objects</a></li>
<li class="chapter" data-level="4.5" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>4.5</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gene-sets.html"><a href="gene-sets.html"><i class="fa fa-check"></i><b>5</b> Gene sets</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>5.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="5.2" data-path="gene-sets.html"><a href="gene-sets.html#mapping-from-entrez-gene-identiifiers-to-go-identifiers"><i class="fa fa-check"></i><b>5.2</b> Mapping from (entrez) gene identiifiers to GO identifiers</a></li>
<li class="chapter" data-level="5.3" data-path="gene-sets.html"><a href="gene-sets.html#using-go-gene-ontology-annotations-to-group-genes-by-biological-process"><i class="fa fa-check"></i><b>5.3</b> Using GO (gene ontology) annotations to group genes by biological process</a></li>
<li class="chapter" data-level="5.4" data-path="gene-sets.html"><a href="gene-sets.html#more-flexible-gene-set-annotation-using-biocset"><i class="fa fa-check"></i><b>5.4</b> More flexible gene set annotation using <span>BiocSet</span></a></li>
<li class="chapter" data-level="5.5" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>5.5</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html"><i class="fa fa-check"></i><b>6</b> (Advanced) Working with SQL databases</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>6.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="6.2" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html#using-dbplyr-to-access-the-ensdb.hsapiens.v75-database"><i class="fa fa-check"></i><b>6.2</b> Using <span>dbplyr</span> to access the [EnsDb.Hsapiens.v75] database</a></li>
<li class="chapter" data-level="6.3" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html#working-with-database-tables"><i class="fa fa-check"></i><b>6.3</b> Working with database tables</a></li>
<li class="chapter" data-level="6.4" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>6.4</b> Saving and session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioconductor Annotation Resources</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="advanced-working-with-sql-databases" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> (Advanced) Working with SQL databases</h1>
<div id="what-do-we-want-to-do" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> What do we want to do?</h2>
<p>Return to our EnsDb package</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="advanced-working-with-sql-databases.html#cb127-1" aria-hidden="true"></a><span class="kw">library</span>(EnsDb.Hsapiens.v75)</span>
<span id="cb127-2"><a href="advanced-working-with-sql-databases.html#cb127-2" aria-hidden="true"></a>EnsDb.Hsapiens.v75</span></code></pre></div>
<pre><code>## EnsDb for Ensembl:
## |Backend: SQLite
## |Db type: EnsDb
## |Type of Gene ID: Ensembl Gene ID
## |Supporting package: ensembldb
## |Db created by: ensembldb package from Bioconductor
## |script_version: 0.3.0
## |Creation time: Thu May 18 09:15:45 2017
## |ensembl_version: 75
## |ensembl_host: localhost
## |Organism: homo_sapiens
## |taxonomy_id: 9606
## |genome_build: GRCh37
## |DBSCHEMAVERSION: 2.0
## | No. of genes: 64102.
## | No. of transcripts: 215647.
## |Protein data available.</code></pre>
<p>Notice the second line of output above</p>
<pre><code>|Backend: SQLite</code></pre>
<p>The package actually ‘wraps’ a SQLite data base. Databases are very useful for
managing large amounts of data, especially when the data consist of several
‘tables’ that are related to one another in various ways.</p>
<p>The <a href="https://cran.r-project.org/package=dbplyr">dbplyr</a> package provides a very convenient way to work with databases,
especially if you know a little R and ‘tidyverse’, but only some SQL.</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="advanced-working-with-sql-databases.html#cb130-1" aria-hidden="true"></a><span class="kw">library</span>(dbplyr)</span></code></pre></div>
</div>
<div id="using-dbplyr-to-access-the-ensdb.hsapiens.v75-database" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Using <a href="https://cran.r-project.org/package=dbplyr">dbplyr</a> to access the [EnsDb.Hsapiens.v75] database</h2>
<p><em>R</em> connects to SQL databases use ‘connections’. Various packages provide
interfaces that connect to a number of different databases, e.g., <a href="https://cran.r-project.org/package=RSQLite">RSQLite</a>, <a href="https://cran.r-project.org/package=RMariaDB">RMariaDB</a>, … The databases can be local, or hosted on a server.</p>
<p>We obtain a connection to the database underlying our EnsDb object using
<code>dbconn()</code>; usually one would use the <code>dbConnect()</code> method defined in the corresponding</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="advanced-working-with-sql-databases.html#cb131-1" aria-hidden="true"></a><span class="kw">dbconn</span>(EnsDb.Hsapiens.v75)</span></code></pre></div>
<pre><code>## &lt;SQLiteConnection&gt;
##   Path: /Users/ma38727/Library/R/4.0/Bioc/3.11/library/EnsDb.Hsapiens.v75/extdata/EnsDb.Hsapiens.v75.sqlite
##   Extensions: TRUE</code></pre>
<p>Use <code>src_dbi()</code> to access the database connection from dbplyr</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="advanced-working-with-sql-databases.html#cb133-1" aria-hidden="true"></a>tbls &lt;-<span class="st"> </span><span class="kw">src_dbi</span>( <span class="kw">dbconn</span>( EnsDb.Hsapiens.v75 ) )</span>
<span id="cb133-2"><a href="advanced-working-with-sql-databases.html#cb133-2" aria-hidden="true"></a>tbls</span></code></pre></div>
<pre><code>## src:  sqlite 3.30.1 [/Users/ma38727/Library/R/4.0/Bioc/3.11/library/EnsDb.Hsapiens.v75/extdata/EnsDb.Hsapiens.v75.sqlite]
## tbls: chromosome, entrezgene, exon, gene, metadata, protein, protein_domain,
##   tx, tx2exon, uniprot</code></pre>
<p>The output shows the location of the sqlite database, as well as the tables
present in the database.</p>
</div>
<div id="working-with-database-tables" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Working with database tables</h2>
<p>Many database operations are very straight-forward with dbplyr. Attach the
<a href="https://cran.r-project.org/package=dply">dplyr</a> package to our session</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="advanced-working-with-sql-databases.html#cb135-1" aria-hidden="true"></a><span class="kw">library</span>(dplyr)</span></code></pre></div>
<p>Select the table of interest</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="advanced-working-with-sql-databases.html#cb136-1" aria-hidden="true"></a><span class="kw">tbl</span>(tbls, <span class="st">&quot;gene&quot;</span>)</span></code></pre></div>
<pre><code>## # Source:   table&lt;gene&gt; [?? x 8]
## # Database: sqlite 3.30.1
## #   [/Users/ma38727/Library/R/4.0/Bioc/3.11/library/EnsDb.Hsapiens.v75/extdata/EnsDb.Hsapiens.v75.sqlite]
##    gene_id gene_name gene_biotype gene_seq_start gene_seq_end seq_name
##    &lt;chr&gt;   &lt;chr&gt;     &lt;chr&gt;                 &lt;int&gt;        &lt;int&gt; &lt;chr&gt;   
##  1 ENSG00… TSPAN6    protein_cod…       99883667     99894988 X       
##  2 ENSG00… TNMD      protein_cod…       99839799     99854882 X       
##  3 ENSG00… DPM1      protein_cod…       49551404     49575092 20      
##  4 ENSG00… SCYL3     protein_cod…      169818772    169863408 1       
##  5 ENSG00… C1orf112  protein_cod…      169631245    169823221 1       
##  6 ENSG00… FGR       protein_cod…       27938575     27961788 1       
##  7 ENSG00… CFH       protein_cod…      196621008    196716634 1       
##  8 ENSG00… FUCA2     protein_cod…      143815948    143832827 6       
##  9 ENSG00… GCLC      protein_cod…       53362139     53481768 6       
## 10 ENSG00… NFYA      protein_cod…       41040684     41067715 6       
## # … with more rows, and 2 more variables: seq_strand &lt;int&gt;,
## #   seq_coord_system &lt;chr&gt;</code></pre>
<p>Note that, unlike the output of an in-memory <code>tibble()</code>, the output does not
display the number of rows in the database table – dbplyr has retrieved only
enough information from the database to display on the screen.</p>
<p>Apply many standard dplyr verbs, e.g.,</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="advanced-working-with-sql-databases.html#cb138-1" aria-hidden="true"></a><span class="kw">tbl</span>(tbls, <span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb138-2"><a href="advanced-working-with-sql-databases.html#cb138-2" aria-hidden="true"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">count</span>(gene_biotype) <span class="op">%&gt;%</span></span>
<span id="cb138-3"><a href="advanced-working-with-sql-databases.html#cb138-3" aria-hidden="true"></a><span class="st">    </span><span class="kw">arrange</span>(<span class="kw">desc</span>(n))</span></code></pre></div>
<pre><code>## # Source:     lazy query [?? x 2]
## # Database:   sqlite 3.30.1
## #   [/Users/ma38727/Library/R/4.0/Bioc/3.11/library/EnsDb.Hsapiens.v75/extdata/EnsDb.Hsapiens.v75.sqlite]
## # Ordered by: desc(n)
##    gene_biotype             n
##    &lt;chr&gt;                &lt;int&gt;
##  1 protein_coding       22810
##  2 pseudogene           15583
##  3 lincRNA               7340
##  4 antisense             5485
##  5 miRNA                 3361
##  6 misc_RNA              2174
##  7 snRNA                 2067
##  8 snoRNA                1549
##  9 processed_transcript   819
## 10 sense_intronic         767
## # … with more rows</code></pre>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="advanced-working-with-sql-databases.html#cb140-1" aria-hidden="true"></a><span class="kw">tbl</span>(tbls, <span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb140-2"><a href="advanced-working-with-sql-databases.html#cb140-2" aria-hidden="true"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(gene_biotype <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;lincRNA&quot;</span>))</span></code></pre></div>
<pre><code>## # Source:   lazy query [?? x 8]
## # Database: sqlite 3.30.1
## #   [/Users/ma38727/Library/R/4.0/Bioc/3.11/library/EnsDb.Hsapiens.v75/extdata/EnsDb.Hsapiens.v75.sqlite]
##    gene_id gene_name gene_biotype gene_seq_start gene_seq_end seq_name
##    &lt;chr&gt;   &lt;chr&gt;     &lt;chr&gt;                 &lt;int&gt;        &lt;int&gt; &lt;chr&gt;   
##  1 ENSG00… TPTEP1    lincRNA            17082777     17179632 22      
##  2 ENSG00… RP11-153… lincRNA            23334268     23404433 12      
##  3 ENSG00… LINC00544 lincRNA            30510042     30524625 13      
##  4 ENSG00… RP11-467… lincRNA            61513965     61525127 11      
##  5 ENSG00… LINC00029 lincRNA            61665569     61668380 20      
##  6 ENSG00… TTTY1B    lincRNA             6258472      6279605 Y       
##  7 ENSG00… TTTY1     lincRNA             9590765      9611898 Y       
##  8 ENSG00… TTTY6     lincRNA            24585740     24587605 Y       
##  9 ENSG00… LINC00470 lincRNA             1254384      1408345 18      
## 10 ENSG00… RP11-84C… lincRNA            21387500     21388458 14      
## # … with more rows, and 2 more variables: seq_strand &lt;int&gt;,
## #   seq_coord_system &lt;chr&gt;</code></pre>
<p>Here’s the ‘tx’ table, which includes the relationship between transcript and
gene</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="advanced-working-with-sql-databases.html#cb142-1" aria-hidden="true"></a><span class="kw">tbl</span>(tbls, <span class="st">&quot;tx&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb142-2"><a href="advanced-working-with-sql-databases.html#cb142-2" aria-hidden="true"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_id, tx_id)</span></code></pre></div>
<pre><code>## # Source:   lazy query [?? x 2]
## # Database: sqlite 3.30.1
## #   [/Users/ma38727/Library/R/4.0/Bioc/3.11/library/EnsDb.Hsapiens.v75/extdata/EnsDb.Hsapiens.v75.sqlite]
##    gene_id         tx_id          
##    &lt;chr&gt;           &lt;chr&gt;          
##  1 ENSG00000000003 ENST00000373020
##  2 ENSG00000000003 ENST00000496771
##  3 ENSG00000000003 ENST00000494424
##  4 ENSG00000000005 ENST00000373031
##  5 ENSG00000000005 ENST00000485971
##  6 ENSG00000000419 ENST00000371588
##  7 ENSG00000000419 ENST00000466152
##  8 ENSG00000000419 ENST00000371582
##  9 ENSG00000000419 ENST00000494752
## 10 ENSG00000000419 ENST00000371584
## # … with more rows</code></pre>
<p>And the ‘protein’ table, which relates transcript to protien</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="advanced-working-with-sql-databases.html#cb144-1" aria-hidden="true"></a><span class="kw">tbl</span>(tbls, <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## # Source:   table&lt;protein&gt; [?? x 3]
## # Database: sqlite 3.30.1
## #   [/Users/ma38727/Library/R/4.0/Bioc/3.11/library/EnsDb.Hsapiens.v75/extdata/EnsDb.Hsapiens.v75.sqlite]
##    tx_id       protein_id    protein_sequence                                   
##    &lt;chr&gt;       &lt;chr&gt;         &lt;chr&gt;                                              
##  1 ENST000003… ENSP00000362… MASPSRRLQTKPVITCFKSVLLIYTFIFWITGVILLAVGIWGKVSLENYF…
##  2 ENST000003… ENSP00000362… MAKNPPENCEDCHILNAEAFKSKKICKSLKICGLVFGILALTLIVLFWGS…
##  3 ENST000003… ENSP00000360… MASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLPLIVWLLVKS…
##  4 ENST000003… ENSP00000360… MASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLPLIVWLLVKS…
##  5 ENST000003… ENSP00000360… XASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLPLIVWLLVKS…
##  6 ENST000004… ENSP00000394… MASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLPLIVWLLVKS…
##  7 ENST000003… ENSP00000360… MASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLPLIVWLLVKS…
##  8 ENST000003… ENSP00000356… MGSENSALKSYTLREPPFTLPSGLAVYPAVLQDGKFASVFVYKRENEDKV…
##  9 ENST000003… ENSP00000356… MGSENSALKSYTLREPPFTLPSGLAVYPAVLQDGKFASVFVYKRENEDKV…
## 10 ENST000004… ENSP00000407… MGSENSALKSYTLREPPFTLPSGLAVYPAVLQDGKFASVFVYKRENEDKV…
## # … with more rows</code></pre>
<p>If we wished to know the protien sequences produced by different genes, we
could form a join between these two tables</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="advanced-working-with-sql-databases.html#cb146-1" aria-hidden="true"></a><span class="kw">left_join</span>(</span>
<span id="cb146-2"><a href="advanced-working-with-sql-databases.html#cb146-2" aria-hidden="true"></a>    <span class="kw">tbl</span>(tbls, <span class="st">&quot;protein&quot;</span>),</span>
<span id="cb146-3"><a href="advanced-working-with-sql-databases.html#cb146-3" aria-hidden="true"></a>    <span class="kw">tbl</span>(tbls, <span class="st">&quot;tx&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb146-4"><a href="advanced-working-with-sql-databases.html#cb146-4" aria-hidden="true"></a><span class="st">        </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_id, tx_id),</span>
<span id="cb146-5"><a href="advanced-working-with-sql-databases.html#cb146-5" aria-hidden="true"></a>    <span class="dt">by =</span> <span class="st">&quot;tx_id&quot;</span></span>
<span id="cb146-6"><a href="advanced-working-with-sql-databases.html#cb146-6" aria-hidden="true"></a>)</span></code></pre></div>
<pre><code>## # Source:   lazy query [?? x 4]
## # Database: sqlite 3.30.1
## #   [/Users/ma38727/Library/R/4.0/Bioc/3.11/library/EnsDb.Hsapiens.v75/extdata/EnsDb.Hsapiens.v75.sqlite]
##    tx_id      protein_id   protein_sequence                           gene_id   
##    &lt;chr&gt;      &lt;chr&gt;        &lt;chr&gt;                                      &lt;chr&gt;     
##  1 ENST00000… ENSP0000036… MASPSRRLQTKPVITCFKSVLLIYTFIFWITGVILLAVGIW… ENSG00000…
##  2 ENST00000… ENSP0000036… MAKNPPENCEDCHILNAEAFKSKKICKSLKICGLVFGILAL… ENSG00000…
##  3 ENST00000… ENSP0000036… MASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLP… ENSG00000…
##  4 ENST00000… ENSP0000036… MASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLP… ENSG00000…
##  5 ENST00000… ENSP0000036… XASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLP… ENSG00000…
##  6 ENST00000… ENSP0000039… MASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLP… ENSG00000…
##  7 ENST00000… ENSP0000036… MASLEVSRSPRRSRRELEVRSPRQNKYSVLLPTYNERENLP… ENSG00000…
##  8 ENST00000… ENSP0000035… MGSENSALKSYTLREPPFTLPSGLAVYPAVLQDGKFASVFV… ENSG00000…
##  9 ENST00000… ENSP0000035… MGSENSALKSYTLREPPFTLPSGLAVYPAVLQDGKFASVFV… ENSG00000…
## 10 ENST00000… ENSP0000040… MGSENSALKSYTLREPPFTLPSGLAVYPAVLQDGKFASVFV… ENSG00000…
## # … with more rows</code></pre>
<p>Knowledge of dplyr verbs coupled with basic knowledge of SQL databases provide
very useful tools for working with large, relationional data.</p>
</div>
<div id="saving-and-session-information" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Saving and session information</h2>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="advanced-working-with-sql-databases.html#cb148-1" aria-hidden="true"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.2 Patched (2020-09-15 r79213)
## Platform: x86_64-apple-darwin17.7.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS:   /Users/ma38727/bin/R-4-0-branch/lib/libRblas.dylib
## LAPACK: /Users/ma38727/bin/R-4-0-branch/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] BiocSet_1.2.2               AnnotationHub_2.20.2       
##  [3] BiocFileCache_1.12.1        dbplyr_1.4.4               
##  [5] DESeq2_1.28.1               airway_1.8.0               
##  [7] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
##  [9] matrixStats_0.56.0          org.Hs.eg.db_3.11.4        
## [11] GO.db_3.11.4                dplyr_1.0.2                
## [13] EnsDb.Hsapiens.v75_2.99.0   ensembldb_2.12.1           
## [15] AnnotationFilter_1.12.0     GenomicFeatures_1.40.1     
## [17] AnnotationDbi_1.50.3        Biobase_2.48.0             
## [19] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
## [21] IRanges_2.22.2              S4Vectors_0.26.1           
## [23] BiocGenerics_0.34.0        
## 
## loaded via a namespace (and not attached):
##  [1] ProtGenerics_1.20.0           bitops_1.0-6                 
##  [3] bit64_4.0.5                   RColorBrewer_1.1-2           
##  [5] progress_1.2.2                httr_1.4.2                   
##  [7] tools_4.0.2                   utf8_1.1.4                   
##  [9] R6_2.4.1                      DBI_1.1.0                    
## [11] lazyeval_0.2.2                colorspace_1.4-1             
## [13] tidyselect_1.1.0              prettyunits_1.1.1            
## [15] bit_4.0.4                     curl_4.3                     
## [17] compiler_4.0.2                cli_2.0.2                    
## [19] rtracklayer_1.48.0            bookdown_0.20                
## [21] scales_1.1.1                  genefilter_1.70.0            
## [23] askpass_1.1                   rappdirs_0.3.1               
## [25] stringr_1.4.0                 digest_0.6.25                
## [27] Rsamtools_2.4.0               rmarkdown_2.3                
## [29] XVector_0.28.0                pkgconfig_2.0.3              
## [31] htmltools_0.5.0               fastmap_1.0.1                
## [33] rlang_0.4.7                   rstudioapi_0.11              
## [35] RSQLite_2.2.0                 shiny_1.5.0                  
## [37] generics_0.0.2                BiocParallel_1.22.0          
## [39] RCurl_1.98-1.2                magrittr_1.5                 
## [41] GenomeInfoDbData_1.2.3        Matrix_1.2-18                
## [43] fansi_0.4.1                   Rcpp_1.0.5                   
## [45] munsell_0.5.0                 lifecycle_0.2.0              
## [47] stringi_1.4.6                 yaml_2.2.1                   
## [49] zlibbioc_1.34.0               plyr_1.8.6                   
## [51] grid_4.0.2                    blob_1.2.1                   
## [53] promises_1.1.1                crayon_1.3.4                 
## [55] lattice_0.20-41               Biostrings_2.56.0            
## [57] splines_4.0.2                 annotate_1.66.0              
## [59] KEGGREST_1.28.0               hms_0.5.3                    
## [61] locfit_1.5-9.4                knitr_1.29                   
## [63] pillar_1.4.6                  codetools_0.2-16             
## [65] geneplotter_1.66.0            biomaRt_2.44.1               
## [67] XML_3.99-0.5                  glue_1.4.2                   
## [69] BiocVersion_3.11.1            evaluate_0.14                
## [71] BiocManager_1.30.10           png_0.1-7                    
## [73] httpuv_1.5.4                  vctrs_0.3.4                  
## [75] gtable_0.3.0                  openssl_1.4.2                
## [77] purrr_0.3.4                   assertthat_0.2.1             
## [79] ggplot2_3.3.2                 xfun_0.16                    
## [81] mime_0.9                      xtable_1.8-4                 
## [83] later_1.1.0.1                 survival_3.2-3               
## [85] tibble_3.0.3                  GenomicAlignments_1.24.0     
## [87] memoise_1.1.0                 ellipsis_0.3.1               
## [89] interactiveDisplayBase_1.26.3</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gene-sets.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-AdvancedDatabase.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to Bioconductor Annotation Resources.pdf", "Introduction to Bioconductor Annotation Resources.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
