<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Bulk RNA Sequencing | Bioconductor Annotation Resources</title>
  <meta name="description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Bulk RNA Sequencing | Bioconductor Annotation Resources" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Bulk RNA Sequencing | Bioconductor Annotation Resources" />
  
  <meta name="twitter:description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  

<meta name="author" content="Martin Morgan" />


<meta name="date" content="2020-09-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="gene-symbol-mapping.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor Annotation Resources</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#plan"><i class="fa fa-check"></i><b>1.1</b> Plan</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#bioconductor"><i class="fa fa-check"></i><b>1.2</b> <em>Bioconductor</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html"><i class="fa fa-check"></i><b>2</b> Bulk RNA Sequencing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#rna-seq-differential-expression-data-in-r-bioconductor"><i class="fa fa-check"></i><b>2.1</b> RNA-seq differential expression data in <em>R</em> / <em>Bioconductor</em></a></li>
<li class="chapter" data-level="2.2" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#a-quick-analysis"><i class="fa fa-check"></i><b>2.2</b> A quick analysis</a></li>
<li class="chapter" data-level="2.3" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>2.3</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html"><i class="fa fa-check"></i><b>3</b> Gene symbol mapping</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>3.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="3.2" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#finding-annotation-resources"><i class="fa fa-check"></i><b>3.2</b> Finding annotation resources</a></li>
<li class="chapter" data-level="3.3" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#mapping-between-identiifiers"><i class="fa fa-check"></i><b>3.3</b> Mapping between identiifiers</a></li>
<li class="chapter" data-level="3.4" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>3.4</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html"><i class="fa fa-check"></i><b>4</b> Discovering genomic coordinates</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>4.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="4.2" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#genomic-ranges"><i class="fa fa-check"></i><b>4.2</b> Genomic ranges</a></li>
<li class="chapter" data-level="4.3" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#mapping-from-identifiers-to-ranges"><i class="fa fa-check"></i><b>4.3</b> Mapping from identifiers to ranges</a></li>
<li class="chapter" data-level="4.4" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#flexible-queries-on-ensdb-objects"><i class="fa fa-check"></i><b>4.4</b> Flexible queries on EnsDb objects</a></li>
<li class="chapter" data-level="4.5" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>4.5</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gene-sets.html"><a href="gene-sets.html"><i class="fa fa-check"></i><b>5</b> Gene sets</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>5.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="5.2" data-path="gene-sets.html"><a href="gene-sets.html#mapping-from-entrez-gene-identiifiers-to-go-identifiers"><i class="fa fa-check"></i><b>5.2</b> Mapping from (entrez) gene identiifiers to GO identifiers</a></li>
<li class="chapter" data-level="5.3" data-path="gene-sets.html"><a href="gene-sets.html#using-go-gene-ontology-annotations-to-group-genes-by-biological-process"><i class="fa fa-check"></i><b>5.3</b> Using GO (gene ontology) annotations to group genes by biological process</a></li>
<li class="chapter" data-level="5.4" data-path="gene-sets.html"><a href="gene-sets.html#more-flexible-gene-set-annotation-using-biocset"><i class="fa fa-check"></i><b>5.4</b> More flexible gene set annotation using <span>BiocSet</span></a></li>
<li class="chapter" data-level="5.5" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>5.5</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioconductor Annotation Resources</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="BulkRNASeq" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Bulk RNA Sequencing</h1>
<div id="rna-seq-differential-expression-data-in-r-bioconductor" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> RNA-seq differential expression data in <em>R</em> / <em>Bioconductor</em></h2>
<p>Bulk RNA sequencing starts by extracting tissue from samples, and then measuring
‘gene expression’ for many 1000’s of genes in each sample. The [airway][] package
is an example of bulk RNA-seq data, processed (see the airway <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/airway/inst/doc/airway.html">vignette</a>,
available from the package landing page) to a ‘SummarizedExperiment’ for use in <em>Bioconductor</em>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="BulkRNASeq.html#cb3-1" aria-hidden="true"></a><span class="kw">library</span>(airway)</span>
<span id="cb3-2"><a href="BulkRNASeq.html#cb3-2" aria-hidden="true"></a><span class="kw">data</span>(airway)</span>
<span id="cb3-3"><a href="BulkRNASeq.html#cb3-3" aria-hidden="true"></a>airway</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 64102 8 
## metadata(1): &#39;&#39;
## assays(1): counts
## rownames(64102): ENSG00000000003 ENSG00000000005 ... LRG_98 LRG_99
## rowData names(0):
## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<p>The essence of an RNA-seq differential expression expression is a matrix of
counts of expression of individual genes in each sample. This data is available
using the accessor <code>assay()</code></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="BulkRNASeq.html#cb5-1" aria-hidden="true"></a><span class="kw">dim</span>(<span class="kw">assay</span>(airway))</span></code></pre></div>
<pre><code>## [1] 64102     8</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="BulkRNASeq.html#cb7-1" aria-hidden="true"></a><span class="kw">head</span>(<span class="kw">assay</span>(airway))</span></code></pre></div>
<pre><code>##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
## ENSG00000000003        679        448        873        408       1138
## ENSG00000000005          0          0          0          0          0
## ENSG00000000419        467        515        621        365        587
## ENSG00000000457        260        211        263        164        245
## ENSG00000000460         60         55         40         35         78
## ENSG00000000938          0          0          2          0          1
##                 SRR1039517 SRR1039520 SRR1039521
## ENSG00000000003       1047        770        572
## ENSG00000000005          0          0          0
## ENSG00000000419        799        417        508
## ENSG00000000457        331        233        229
## ENSG00000000460         63         76         60
## ENSG00000000938          0          0          0</code></pre>
<p>Usually the samples are taken from an <em>experimental design</em>. The samples might be
from an actual experiment, or from, e.g., different tissues, etc.</p>
<p>Here’s the experimental design:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="BulkRNASeq.html#cb9-1" aria-hidden="true"></a><span class="kw">colData</span>(airway)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 9 columns
##            SampleName     cell      dex    albut        Run avgLength
##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;
## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126
## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126
## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126
## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87
## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120
## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126
## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101
## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98
##            Experiment    Sample    BioSample
##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;
## SRR1039508  SRX384345 SRS508568 SAMN02422669
## SRR1039509  SRX384346 SRS508567 SAMN02422675
## SRR1039512  SRX384349 SRS508571 SAMN02422678
## SRR1039513  SRX384350 SRS508572 SAMN02422670
## SRR1039516  SRX384353 SRS508575 SAMN02422682
## SRR1039517  SRX384354 SRS508576 SAMN02422673
## SRR1039520  SRX384357 SRS508579 SAMN02422683
## SRR1039521  SRX384358 SRS508580 SAMN02422677</code></pre>
<p>There are 8 samples (‘Sample’) from four cell lines (‘cell’), exposed to two treatments of
dexamethasone (‘dex’)</p>
<p>When I explore data I sometimes like to use the ‘tidy’ approach</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="BulkRNASeq.html#cb11-1" aria-hidden="true"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb11-2"><a href="BulkRNASeq.html#cb11-2" aria-hidden="true"></a>design &lt;-</span>
<span id="cb11-3"><a href="BulkRNASeq.html#cb11-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">colData</span>(airway) <span class="op">%&gt;%</span></span>
<span id="cb11-4"><a href="BulkRNASeq.html#cb11-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span></span>
<span id="cb11-5"><a href="BulkRNASeq.html#cb11-5" aria-hidden="true"></a><span class="st">  </span><span class="kw">select</span>(Sample, cell, dex)</span>
<span id="cb11-6"><a href="BulkRNASeq.html#cb11-6" aria-hidden="true"></a>design</span></code></pre></div>
<pre><code>## # A tibble: 8 x 3
##   Sample    cell    dex  
##   &lt;fct&gt;     &lt;fct&gt;   &lt;fct&gt;
## 1 SRS508568 N61311  untrt
## 2 SRS508567 N61311  trt  
## 3 SRS508571 N052611 untrt
## 4 SRS508572 N052611 trt  
## 5 SRS508575 N080611 untrt
## 6 SRS508576 N080611 trt  
## 7 SRS508579 N061011 untrt
## 8 SRS508580 N061011 trt</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="BulkRNASeq.html#cb13-1" aria-hidden="true"></a>design <span class="op">%&gt;%</span></span>
<span id="cb13-2"><a href="BulkRNASeq.html#cb13-2" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">count</span>(dex)</span></code></pre></div>
<pre><code>## # A tibble: 2 x 2
##   dex       n
##   &lt;fct&gt; &lt;int&gt;
## 1 trt       4
## 2 untrt     4</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="BulkRNASeq.html#cb15-1" aria-hidden="true"></a>design <span class="op">%&gt;%</span></span>
<span id="cb15-2"><a href="BulkRNASeq.html#cb15-2" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">count</span>(cell)</span></code></pre></div>
<pre><code>## # A tibble: 4 x 2
##   cell        n
##   &lt;fct&gt;   &lt;int&gt;
## 1 N052611     2
## 2 N061011     2
## 3 N080611     2
## 4 N61311      2</code></pre>
<p>Today we’ll focus on the rows (genes) of the RNASeq experiment. In the airway
data, the ‘rowData’ are very simple and consist just of <a href="https://ensemblgenomes.org/">Ensembl</a> gene identifiers</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="BulkRNASeq.html#cb17-1" aria-hidden="true"></a><span class="kw">rowData</span>(airway) <span class="op">%&gt;%</span></span>
<span id="cb17-2"><a href="BulkRNASeq.html#cb17-2" aria-hidden="true"></a><span class="st">  </span><span class="kw">as_tibble</span>(<span class="dt">rownames =</span> <span class="st">&quot;Gene&quot;</span>)</span></code></pre></div>
<pre><code>## # A tibble: 64,102 x 1
##    Gene           
##    &lt;chr&gt;          
##  1 ENSG00000000003
##  2 ENSG00000000005
##  3 ENSG00000000419
##  4 ENSG00000000457
##  5 ENSG00000000460
##  6 ENSG00000000938
##  7 ENSG00000000971
##  8 ENSG00000001036
##  9 ENSG00000001084
## 10 ENSG00000001167
## # … with 64,092 more rows</code></pre>
</div>
<div id="a-quick-analysis" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> A quick analysis</h2>
<p>It’s beyond the scope of today’s lecture, but let’s do a quick analysis of
differenetial expression in the airway dataset.</p>
<p>Start by loading the [DESeq2][] package</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="BulkRNASeq.html#cb19-1" aria-hidden="true"></a><span class="kw">library</span>(DESeq2)</span></code></pre></div>
<p>Prepare the data by ‘re-leveling’ the ‘dex’ treatment variable, and describing
the experirmental design</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="BulkRNASeq.html#cb20-1" aria-hidden="true"></a><span class="kw">colData</span>(airway)<span class="op">$</span>dex &lt;-<span class="st"> </span><span class="kw">relevel</span>(<span class="kw">colData</span>(airway)<span class="op">$</span>dex, <span class="st">&quot;untrt&quot;</span>)</span>
<span id="cb20-2"><a href="BulkRNASeq.html#cb20-2" aria-hidden="true"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSet</span>(airway, <span class="op">~</span><span class="st"> </span>cell <span class="op">+</span><span class="st"> </span>dex)</span></code></pre></div>
<p>Perform the differential expression analysis…</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="BulkRNASeq.html#cb21-1" aria-hidden="true"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<p>…and summarize results in terms average expression (‘baseMean’), adjusted
P-value (‘padj’), etc.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="BulkRNASeq.html#cb28-1" aria-hidden="true"></a><span class="kw">results</span>(dds)</span></code></pre></div>
<pre><code>## log2 fold change (MLE): dex trt vs untrt 
## Wald test p-value: dex trt vs untrt 
## DataFrame with 64102 rows and 6 columns
##                  baseMean log2FoldChange     lfcSE      stat      pvalue
##                 &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
## ENSG00000000003  708.6022     -0.3812539  0.100654 -3.787751 0.000152017
## ENSG00000000005    0.0000             NA        NA        NA          NA
## ENSG00000000419  520.2979      0.2068127  0.112219  1.842944 0.065337210
## ENSG00000000457  237.1630      0.0379206  0.143445  0.264357 0.791504963
## ENSG00000000460   57.9326     -0.0881677  0.287142 -0.307053 0.758803336
## ...                   ...            ...       ...       ...         ...
## LRG_94                  0             NA        NA        NA          NA
## LRG_96                  0             NA        NA        NA          NA
## LRG_97                  0             NA        NA        NA          NA
## LRG_98                  0             NA        NA        NA          NA
## LRG_99                  0             NA        NA        NA          NA
##                       padj
##                  &lt;numeric&gt;
## ENSG00000000003 0.00128364
## ENSG00000000005         NA
## ENSG00000000419 0.19654584
## ENSG00000000457 0.91145800
## ENSG00000000460 0.89503445
## ...                    ...
## LRG_94                  NA
## LRG_96                  NA
## LRG_97                  NA
## LRG_98                  NA
## LRG_99                  NA</code></pre>
<p>For convenience, add the results to the <code>rowData()</code> of the airway dataset</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="BulkRNASeq.html#cb30-1" aria-hidden="true"></a><span class="kw">rowData</span>(airway) &lt;-<span class="st"> </span><span class="kw">results</span>(dds)</span></code></pre></div>
</div>
<div id="saving-and-session-information" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Saving and session information</h2>
<p>Save the updated object for use in the next chapter.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="BulkRNASeq.html#cb31-1" aria-hidden="true"></a><span class="kw">saveRDS</span>(airway, <span class="st">&quot;airway_with_results.Rda&quot;</span>)</span></code></pre></div>
<p>Use the <code>sessionInfo()</code> command to record the precise software versions we used.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="BulkRNASeq.html#cb32-1" aria-hidden="true"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.2 Patched (2020-09-15 r79213)
## Platform: x86_64-apple-darwin17.7.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS:   /Users/ma38727/bin/R-4-0-branch/lib/libRblas.dylib
## LAPACK: /Users/ma38727/bin/R-4-0-branch/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] DESeq2_1.28.1               airway_1.8.0               
##  [3] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
##  [5] matrixStats_0.56.0          org.Hs.eg.db_3.11.4        
##  [7] GO.db_3.11.4                dplyr_1.0.2                
##  [9] EnsDb.Hsapiens.v75_2.99.0   ensembldb_2.12.1           
## [11] AnnotationFilter_1.12.0     GenomicFeatures_1.40.1     
## [13] AnnotationDbi_1.50.3        Biobase_2.48.0             
## [15] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
## [17] IRanges_2.22.2              S4Vectors_0.26.1           
## [19] BiocGenerics_0.34.0        
## 
## loaded via a namespace (and not attached):
##  [1] ProtGenerics_1.20.0      bitops_1.0-6             bit64_4.0.5             
##  [4] RColorBrewer_1.1-2       progress_1.2.2           httr_1.4.2              
##  [7] tools_4.0.2              utf8_1.1.4               R6_2.4.1                
## [10] DBI_1.1.0                lazyeval_0.2.2           colorspace_1.4-1        
## [13] tidyselect_1.1.0         prettyunits_1.1.1        bit_4.0.4               
## [16] curl_4.3                 compiler_4.0.2           cli_2.0.2               
## [19] rtracklayer_1.48.0       bookdown_0.20            scales_1.1.1            
## [22] genefilter_1.70.0        askpass_1.1              rappdirs_0.3.1          
## [25] stringr_1.4.0            digest_0.6.25            Rsamtools_2.4.0         
## [28] rmarkdown_2.3            XVector_0.28.0           pkgconfig_2.0.3         
## [31] htmltools_0.5.0          dbplyr_1.4.4             rlang_0.4.7             
## [34] rstudioapi_0.11          RSQLite_2.2.0            generics_0.0.2          
## [37] BiocParallel_1.22.0      RCurl_1.98-1.2           magrittr_1.5            
## [40] GenomeInfoDbData_1.2.3   Matrix_1.2-18            Rcpp_1.0.5              
## [43] munsell_0.5.0            fansi_0.4.1              lifecycle_0.2.0         
## [46] stringi_1.4.6            yaml_2.2.1               zlibbioc_1.34.0         
## [49] BiocFileCache_1.12.1     grid_4.0.2               blob_1.2.1              
## [52] crayon_1.3.4             lattice_0.20-41          Biostrings_2.56.0       
## [55] splines_4.0.2            annotate_1.66.0          hms_0.5.3               
## [58] locfit_1.5-9.4           knitr_1.29               pillar_1.4.6            
## [61] geneplotter_1.66.0       codetools_0.2-16         biomaRt_2.44.1          
## [64] XML_3.99-0.5             glue_1.4.2               evaluate_0.14           
## [67] vctrs_0.3.4              gtable_0.3.0             openssl_1.4.2           
## [70] purrr_0.3.4              assertthat_0.2.1         ggplot2_3.3.2           
## [73] xfun_0.16                xtable_1.8-4             survival_3.2-3          
## [76] tibble_3.0.3             GenomicAlignments_1.24.0 memoise_1.1.0           
## [79] ellipsis_0.3.1</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gene-symbol-mapping.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/01-RNASeq.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to Bioconductor Annotation Resources.pdf", "Introduction to Bioconductor Annotation Resources.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
