<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Discovering genomic coordinates | Bioconductor Annotation Resources</title>
  <meta name="description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Discovering genomic coordinates | Bioconductor Annotation Resources" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Discovering genomic coordinates | Bioconductor Annotation Resources" />
  
  <meta name="twitter:description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  

<meta name="author" content="Martin Morgan" />


<meta name="date" content="2020-09-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gene-symbol-mapping.html"/>
<link rel="next" href="gene-sets.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor Annotation Resources</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#plan"><i class="fa fa-check"></i><b>1.1</b> Plan</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#bioconductor"><i class="fa fa-check"></i><b>1.2</b> <em>Bioconductor</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html"><i class="fa fa-check"></i><b>2</b> Bulk RNA Sequencing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#rna-seq-differential-expression-data-in-r-bioconductor"><i class="fa fa-check"></i><b>2.1</b> RNA-seq differential expression data in <em>R</em> / <em>Bioconductor</em></a></li>
<li class="chapter" data-level="2.2" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#a-quick-analysis"><i class="fa fa-check"></i><b>2.2</b> A quick analysis</a></li>
<li class="chapter" data-level="2.3" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>2.3</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html"><i class="fa fa-check"></i><b>3</b> Gene symbol mapping</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>3.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="3.2" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#finding-annotation-resources"><i class="fa fa-check"></i><b>3.2</b> Finding annotation resources</a></li>
<li class="chapter" data-level="3.3" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#mapping-between-identiifiers"><i class="fa fa-check"></i><b>3.3</b> Mapping between identiifiers</a></li>
<li class="chapter" data-level="3.4" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>3.4</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html"><i class="fa fa-check"></i><b>4</b> Discovering genomic coordinates</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>4.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="4.2" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#genomic-ranges"><i class="fa fa-check"></i><b>4.2</b> Genomic ranges</a></li>
<li class="chapter" data-level="4.3" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#mapping-from-identifiers-to-ranges"><i class="fa fa-check"></i><b>4.3</b> Mapping from identifiers to ranges</a></li>
<li class="chapter" data-level="4.4" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#flexible-queries-on-ensdb-objects"><i class="fa fa-check"></i><b>4.4</b> Flexible queries on EnsDb objects</a></li>
<li class="chapter" data-level="4.5" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>4.5</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gene-sets.html"><a href="gene-sets.html"><i class="fa fa-check"></i><b>5</b> Gene sets</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>5.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="5.2" data-path="gene-sets.html"><a href="gene-sets.html#mapping-from-entrez-gene-identiifiers-to-go-identifiers"><i class="fa fa-check"></i><b>5.2</b> Mapping from (entrez) gene identiifiers to GO identifiers</a></li>
<li class="chapter" data-level="5.3" data-path="gene-sets.html"><a href="gene-sets.html#using-go-gene-ontology-annotations-to-group-genes-by-biological-process"><i class="fa fa-check"></i><b>5.3</b> Using GO (gene ontology) annotations to group genes by biological process</a></li>
<li class="chapter" data-level="5.4" data-path="gene-sets.html"><a href="gene-sets.html#more-flexible-gene-set-annotation-using-biocset"><i class="fa fa-check"></i><b>5.4</b> More flexible gene set annotation using <span>BiocSet</span></a></li>
<li class="chapter" data-level="5.5" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>5.5</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html"><i class="fa fa-check"></i><b>6</b> (Advanced) Working with SQL databases</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>6.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="6.2" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html#using-dbplyr-to-access-the-ensdb.hsapiens.v75-database"><i class="fa fa-check"></i><b>6.2</b> Using <span>dbplyr</span> to access the [EnsDb.Hsapiens.v75] database</a></li>
<li class="chapter" data-level="6.3" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html#working-with-database-tables"><i class="fa fa-check"></i><b>6.3</b> Working with database tables</a></li>
<li class="chapter" data-level="6.4" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>6.4</b> Saving and session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioconductor Annotation Resources</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="discovering-genomic-coordinates" class="section level1" number="4">
<h1><span class="header-section-number">Chapter 4</span> Discovering genomic coordinates</h1>
<div id="what-do-we-want-to-do" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> What do we want to do?</h2>
<p>Load the annotation and tidy packages, and our work from the last chapter</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="discovering-genomic-coordinates.html#cb70-1" aria-hidden="true"></a><span class="kw">library</span>(EnsDb.Hsapiens.v75)</span>
<span id="cb70-2"><a href="discovering-genomic-coordinates.html#cb70-2" aria-hidden="true"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb70-3"><a href="discovering-genomic-coordinates.html#cb70-3" aria-hidden="true"></a>toptbl &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;toptbl.Rda&quot;</span>)</span>
<span id="cb70-4"><a href="discovering-genomic-coordinates.html#cb70-4" aria-hidden="true"></a>toptbl</span></code></pre></div>
<pre><code>## # A tibble: 64,102 x 9
##    Gene  SYMBOL ENTREZID baseMean log2FoldChange  lfcSE  stat    pvalue
##    &lt;chr&gt; &lt;chr&gt;     &lt;int&gt;    &lt;dbl&gt;          &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;
##  1 ENSG… SPARC…     8404     997.           4.57 0.184   24.9 2.22e-136
##  2 ENSG… CACNB2      783     495.           3.29 0.133   24.7 7.84e-135
##  3 ENSG… DUSP1      1843    3409.           2.95 0.121   24.3 3.67e-130
##  4 ENSG… SAMHD1    25939   12703.           3.77 0.155   24.2 9.58e-130
##  5 ENSG… MAOA       4128    2342.           3.35 0.142   23.7 1.10e-123
##  6 ENSG… GPX3       2878   12286.           3.73 0.166   22.5 4.62e-112
##  7 ENSG… STEAP2   261729    3009.           1.98 0.0900  22.0 5.77e-107
##  8 ENSG… NEXN      91624    5393.           2.04 0.0942  21.6 1.32e-103
##  9 ENSG… MT2A       4502    3656.           2.21 0.106   20.9 2.90e- 97
## 10 ENSG… ADAMT…     9510   30315.           2.35 0.116   20.3 3.26e- 91
## # … with 64,092 more rows, and 1 more variable: padj &lt;dbl&gt;</code></pre>
<p>Can we get information about the genomic coordinates (e.g., chromosome, strand,
start and end nucleotides) of each of these genes?</p>
</div>
<div id="genomic-ranges" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Genomic ranges</h2>
<p><em>Bioconductor</em> uses the [GenomicRanges][] package to represent genomic
coordinates.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="discovering-genomic-coordinates.html#cb72-1" aria-hidden="true"></a><span class="kw">library</span>(GenomicRanges)</span>
<span id="cb72-2"><a href="discovering-genomic-coordinates.html#cb72-2" aria-hidden="true"></a><span class="kw">GRanges</span>(</span>
<span id="cb72-3"><a href="discovering-genomic-coordinates.html#cb72-3" aria-hidden="true"></a>  <span class="kw">c</span>(<span class="st">&quot;chr1:1-10&quot;</span>, <span class="st">&quot;chr2:20-30:+&quot;</span>)</span>
<span id="cb72-4"><a href="discovering-genomic-coordinates.html#cb72-4" aria-hidden="true"></a>)</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr1      1-10      *
##   [2]     chr2     20-30      +
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p>Genomic ranges consist of <code>seqnames</code> (e.g., chromosome, but could be a
contig or other DNA / RNA / amino acid sequence), <code>start</code> and <code>end</code> coordinates,
and an optional <code>strand</code>.</p>
<p>The coordinates are 1-based, closed intervals. So the first range in the example
above starts at the very first nucleotide on chr1, and includes the 10th
nucleotide – there are 11 total nucleotides (1, 2, …, 10) in the range in
total! This differs from some schemes, notably that used in the UCSC genome
browser, where the ranges are 0-based and half-open (the start but not the end
nucelotide is included in the range) and the example above would have been
written ‘chr1:0-10’.</p>
<p>There are a large number of very useful operations that can be performed on
GRanges; see the vignettes in the GenomicRanges package, available on the
package [landing page][GenomicRanges].</p>
</div>
<div id="mapping-from-identifiers-to-ranges" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Mapping from identifiers to ranges</h2>
<p>The coordinates of all genes are included in the EnsDb packages. Use the
<code>genes()</code> accessor to extract the ranges.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="discovering-genomic-coordinates.html#cb74-1" aria-hidden="true"></a><span class="kw">genes</span>(EnsDb.Hsapiens.v75)</span></code></pre></div>
<pre><code>## GRanges object with 64102 ranges and 6 metadata columns:
##                   seqnames            ranges strand |         gene_id
##                      &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
##   ENSG00000223972        1       11869-14412      + | ENSG00000223972
##   ENSG00000227232        1       14363-29806      - | ENSG00000227232
##   ENSG00000243485        1       29554-31109      + | ENSG00000243485
##   ENSG00000237613        1       34554-36081      - | ENSG00000237613
##   ENSG00000268020        1       52473-54936      + | ENSG00000268020
##               ...      ...               ...    ... .             ...
##   ENSG00000224240        Y 28695572-28695890      + | ENSG00000224240
##   ENSG00000227629        Y 28732789-28737748      - | ENSG00000227629
##   ENSG00000237917        Y 28740998-28780799      - | ENSG00000237917
##   ENSG00000231514        Y 28772667-28773306      - | ENSG00000231514
##   ENSG00000235857        Y 59001391-59001635      + | ENSG00000235857
##                     gene_name gene_biotype seq_coord_system      symbol
##                   &lt;character&gt;  &lt;character&gt;      &lt;character&gt; &lt;character&gt;
##   ENSG00000223972     DDX11L1   pseudogene       chromosome     DDX11L1
##   ENSG00000227232      WASH7P   pseudogene       chromosome      WASH7P
##   ENSG00000243485  MIR1302-10      lincRNA       chromosome  MIR1302-10
##   ENSG00000237613     FAM138A      lincRNA       chromosome     FAM138A
##   ENSG00000268020      OR4G4P   pseudogene       chromosome      OR4G4P
##               ...         ...          ...              ...         ...
##   ENSG00000224240     CYCSP49   pseudogene       chromosome     CYCSP49
##   ENSG00000227629  SLC25A15P1   pseudogene       chromosome  SLC25A15P1
##   ENSG00000237917     PARP4P1   pseudogene       chromosome     PARP4P1
##   ENSG00000231514     FAM58CP   pseudogene       chromosome     FAM58CP
##   ENSG00000235857     CTBP2P1   pseudogene       chromosome     CTBP2P1
##                                            entrezid
##                                              &lt;list&gt;
##   ENSG00000223972               100287596,100287102
##   ENSG00000227232                  100287171,653635
##   ENSG00000243485 100422919,100422834,100422831,...
##   ENSG00000237613              654835,645520,641702
##   ENSG00000268020                                NA
##               ...                               ...
##   ENSG00000224240                                NA
##   ENSG00000227629                                NA
##   ENSG00000237917                                NA
##   ENSG00000231514                                NA
##   ENSG00000235857                                NA
##   -------
##   seqinfo: 273 sequences from GRCh37 genome</code></pre>
<p>Note that the GRanges have names corresponding to Ensembl gene identifiers,
so the ranges ordered according to our top table are</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="discovering-genomic-coordinates.html#cb76-1" aria-hidden="true"></a><span class="kw">genes</span>(EnsDb.Hsapiens.v75)[toptbl<span class="op">$</span>Gene]</span></code></pre></div>
<pre><code>## GRanges object with 64102 ranges and 6 metadata columns:
##                   seqnames              ranges strand |         gene_id
##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
##   ENSG00000152583        4   88394487-88452213      - | ENSG00000152583
##   ENSG00000165995       10   18429606-18830798      + | ENSG00000165995
##   ENSG00000120129        5 172195093-172198198      - | ENSG00000120129
##   ENSG00000101347       20   35518632-35580246      - | ENSG00000101347
##   ENSG00000189221        X   43515467-43606068      + | ENSG00000189221
##               ...      ...                 ...    ... .             ...
##            LRG_94       10   72357104-72362531      - |          LRG_94
##            LRG_96       15   55495792-55582001      - |          LRG_96
##            LRG_97       22   37621310-37640305      - |          LRG_97
##            LRG_98       11   36589563-36601312      + |          LRG_98
##            LRG_99       11   36613493-36619812      - |          LRG_99
##                     gene_name   gene_biotype seq_coord_system      symbol
##                   &lt;character&gt;    &lt;character&gt;      &lt;character&gt; &lt;character&gt;
##   ENSG00000152583     SPARCL1 protein_coding       chromosome     SPARCL1
##   ENSG00000165995      CACNB2 protein_coding       chromosome      CACNB2
##   ENSG00000120129       DUSP1 protein_coding       chromosome       DUSP1
##   ENSG00000101347      SAMHD1 protein_coding       chromosome      SAMHD1
##   ENSG00000189221        MAOA protein_coding       chromosome        MAOA
##               ...         ...            ...              ...         ...
##            LRG_94      LRG_94       LRG_gene       chromosome      LRG_94
##            LRG_96      LRG_96       LRG_gene       chromosome      LRG_96
##            LRG_97      LRG_97       LRG_gene       chromosome      LRG_97
##            LRG_98      LRG_98       LRG_gene       chromosome      LRG_98
##            LRG_99      LRG_99       LRG_gene       chromosome      LRG_99
##                   entrezid
##                     &lt;list&gt;
##   ENSG00000152583     8404
##   ENSG00000165995      783
##   ENSG00000120129     1843
##   ENSG00000101347    25939
##   ENSG00000189221     4128
##               ...      ...
##            LRG_94     5551
##            LRG_96     5873
##            LRG_97     5880
##            LRG_98     5896
##            LRG_99     5897
##   -------
##   seqinfo: 273 sequences from GRCh37 genome</code></pre>
<p>It is possible to get addition information from the EnsDb, e.g., the coordinates
of all exons in a gene.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="discovering-genomic-coordinates.html#cb78-1" aria-hidden="true"></a><span class="kw">exonsBy</span>(EnsDb.Hsapiens.v75, <span class="st">&quot;gene&quot;</span>)[toptbl<span class="op">$</span>Gene]</span></code></pre></div>
<pre><code>## GRangesList object of length 64102:
## $ENSG00000152583
## GRanges object with 39 ranges and 1 metadata column:
##        seqnames            ranges strand |         exon_id
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
##    [1]        4 88452130-88452213      - | ENSE00002034086
##    [2]        4 88452130-88452201      - | ENSE00002083738
##    [3]        4 88451697-88451783      - | ENSE00002020252
##    [4]        4 88450197-88450656      - | ENSE00001942085
##    [5]        4 88450197-88450655      - | ENSE00001603831
##    ...      ...               ...    ... .             ...
##   [35]        4 88401524-88401672      - | ENSE00001175678
##   [36]        4 88400582-88400730      - | ENSE00001175672
##   [37]        4 88394487-88394955      - | ENSE00001914146
##   [38]        4 88394570-88394955      - | ENSE00002056886
##   [39]        4 88394500-88394955      - | ENSE00001135423
##   -------
##   seqinfo: 273 sequences from GRCh37 genome
## 
## ...
## &lt;64101 more elements&gt;</code></pre>
</div>
<div id="flexible-queries-on-ensdb-objects" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Flexible queries on EnsDb objects</h2>
<p>The EnsDb objects allow very flexible queries. Suppose we were interested in
the top 6 most differentially expressed gene symbols</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="discovering-genomic-coordinates.html#cb80-1" aria-hidden="true"></a>ids &lt;-<span class="st"> </span><span class="kw">head</span>(toptbl<span class="op">$</span>SYMBOL)</span>
<span id="cb80-2"><a href="discovering-genomic-coordinates.html#cb80-2" aria-hidden="true"></a>ids</span></code></pre></div>
<pre><code>## ENSG00000152583 ENSG00000165995 ENSG00000120129 ENSG00000101347 ENSG00000189221 
##       &quot;SPARCL1&quot;        &quot;CACNB2&quot;         &quot;DUSP1&quot;        &quot;SAMHD1&quot;          &quot;MAOA&quot; 
## ENSG00000211445 
##          &quot;GPX3&quot;</code></pre>
<p>We can write an expression like</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="discovering-genomic-coordinates.html#cb82-1" aria-hidden="true"></a><span class="kw">genes</span>(EnsDb.Hsapiens.v75, <span class="dt">filter =</span> <span class="op">~</span><span class="st"> </span>genename <span class="op">%in%</span><span class="st"> </span>ids)</span></code></pre></div>
<pre><code>## GRanges object with 6 ranges and 6 metadata columns:
##                   seqnames              ranges strand |         gene_id
##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
##   ENSG00000165995       10   18429606-18830798      + | ENSG00000165995
##   ENSG00000101347       20   35518632-35580246      - | ENSG00000101347
##   ENSG00000152583        4   88394487-88452213      - | ENSG00000152583
##   ENSG00000211445        5 150400124-150408554      + | ENSG00000211445
##   ENSG00000120129        5 172195093-172198198      - | ENSG00000120129
##   ENSG00000189221        X   43515467-43606068      + | ENSG00000189221
##                     gene_name   gene_biotype seq_coord_system      symbol
##                   &lt;character&gt;    &lt;character&gt;      &lt;character&gt; &lt;character&gt;
##   ENSG00000165995      CACNB2 protein_coding       chromosome      CACNB2
##   ENSG00000101347      SAMHD1 protein_coding       chromosome      SAMHD1
##   ENSG00000152583     SPARCL1 protein_coding       chromosome     SPARCL1
##   ENSG00000211445        GPX3 protein_coding       chromosome        GPX3
##   ENSG00000120129       DUSP1 protein_coding       chromosome       DUSP1
##   ENSG00000189221        MAOA protein_coding       chromosome        MAOA
##                   entrezid
##                     &lt;list&gt;
##   ENSG00000165995      783
##   ENSG00000101347    25939
##   ENSG00000152583     8404
##   ENSG00000211445     2878
##   ENSG00000120129     1843
##   ENSG00000189221     4128
##   -------
##   seqinfo: 5 sequences from GRCh37 genome</code></pre>
<p>to select just the genomic ranges of interest, and</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="discovering-genomic-coordinates.html#cb84-1" aria-hidden="true"></a><span class="kw">genes</span>(EnsDb.Hsapiens.v75, <span class="dt">filter =</span> <span class="op">~</span><span class="st"> </span>genename <span class="op">%in%</span><span class="st"> </span>ids)[<span class="kw">names</span>(ids)]</span></code></pre></div>
<pre><code>## GRanges object with 6 ranges and 6 metadata columns:
##                   seqnames              ranges strand |         gene_id
##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
##   ENSG00000152583        4   88394487-88452213      - | ENSG00000152583
##   ENSG00000165995       10   18429606-18830798      + | ENSG00000165995
##   ENSG00000120129        5 172195093-172198198      - | ENSG00000120129
##   ENSG00000101347       20   35518632-35580246      - | ENSG00000101347
##   ENSG00000189221        X   43515467-43606068      + | ENSG00000189221
##   ENSG00000211445        5 150400124-150408554      + | ENSG00000211445
##                     gene_name   gene_biotype seq_coord_system      symbol
##                   &lt;character&gt;    &lt;character&gt;      &lt;character&gt; &lt;character&gt;
##   ENSG00000152583     SPARCL1 protein_coding       chromosome     SPARCL1
##   ENSG00000165995      CACNB2 protein_coding       chromosome      CACNB2
##   ENSG00000120129       DUSP1 protein_coding       chromosome       DUSP1
##   ENSG00000101347      SAMHD1 protein_coding       chromosome      SAMHD1
##   ENSG00000189221        MAOA protein_coding       chromosome        MAOA
##   ENSG00000211445        GPX3 protein_coding       chromosome        GPX3
##                   entrezid
##                     &lt;list&gt;
##   ENSG00000152583     8404
##   ENSG00000165995      783
##   ENSG00000120129     1843
##   ENSG00000101347    25939
##   ENSG00000189221     4128
##   ENSG00000211445     2878
##   -------
##   seqinfo: 5 sequences from GRCh37 genome</code></pre>
<p>to place these ranges in ‘top table’ order.</p>
<p>You’ll have an opportunity to explore more about genomic ranges when Dr.
Stephanie Hicks talks on Tuesday, October 6.</p>
</div>
<div id="saving-and-session-information" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Saving and session information</h2>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="discovering-genomic-coordinates.html#cb86-1" aria-hidden="true"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.2 Patched (2020-09-15 r79213)
## Platform: x86_64-apple-darwin17.7.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS:   /Users/ma38727/bin/R-4-0-branch/lib/libRblas.dylib
## LAPACK: /Users/ma38727/bin/R-4-0-branch/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] AnnotationHub_2.20.2        BiocFileCache_1.12.1       
##  [3] dbplyr_1.4.4                DESeq2_1.28.1              
##  [5] airway_1.8.0                SummarizedExperiment_1.18.2
##  [7] DelayedArray_0.14.1         matrixStats_0.56.0         
##  [9] org.Hs.eg.db_3.11.4         GO.db_3.11.4               
## [11] dplyr_1.0.2                 EnsDb.Hsapiens.v75_2.99.0  
## [13] ensembldb_2.12.1            AnnotationFilter_1.12.0    
## [15] GenomicFeatures_1.40.1      AnnotationDbi_1.50.3       
## [17] Biobase_2.48.0              GenomicRanges_1.40.0       
## [19] GenomeInfoDb_1.24.2         IRanges_2.22.2             
## [21] S4Vectors_0.26.1            BiocGenerics_0.34.0        
## 
## loaded via a namespace (and not attached):
##  [1] ProtGenerics_1.20.0           bitops_1.0-6                 
##  [3] bit64_4.0.5                   RColorBrewer_1.1-2           
##  [5] progress_1.2.2                httr_1.4.2                   
##  [7] tools_4.0.2                   utf8_1.1.4                   
##  [9] R6_2.4.1                      DBI_1.1.0                    
## [11] lazyeval_0.2.2                colorspace_1.4-1             
## [13] tidyselect_1.1.0              prettyunits_1.1.1            
## [15] bit_4.0.4                     curl_4.3                     
## [17] compiler_4.0.2                cli_2.0.2                    
## [19] rtracklayer_1.48.0            bookdown_0.20                
## [21] scales_1.1.1                  genefilter_1.70.0            
## [23] askpass_1.1                   rappdirs_0.3.1               
## [25] stringr_1.4.0                 digest_0.6.25                
## [27] Rsamtools_2.4.0               rmarkdown_2.3                
## [29] XVector_0.28.0                pkgconfig_2.0.3              
## [31] htmltools_0.5.0               fastmap_1.0.1                
## [33] rlang_0.4.7                   rstudioapi_0.11              
## [35] RSQLite_2.2.0                 shiny_1.5.0                  
## [37] generics_0.0.2                BiocParallel_1.22.0          
## [39] RCurl_1.98-1.2                magrittr_1.5                 
## [41] GenomeInfoDbData_1.2.3        Matrix_1.2-18                
## [43] Rcpp_1.0.5                    munsell_0.5.0                
## [45] fansi_0.4.1                   lifecycle_0.2.0              
## [47] stringi_1.4.6                 yaml_2.2.1                   
## [49] zlibbioc_1.34.0               grid_4.0.2                   
## [51] blob_1.2.1                    promises_1.1.1               
## [53] crayon_1.3.4                  lattice_0.20-41              
## [55] Biostrings_2.56.0             splines_4.0.2                
## [57] annotate_1.66.0               hms_0.5.3                    
## [59] locfit_1.5-9.4                knitr_1.29                   
## [61] pillar_1.4.6                  geneplotter_1.66.0           
## [63] codetools_0.2-16              biomaRt_2.44.1               
## [65] BiocVersion_3.11.1            XML_3.99-0.5                 
## [67] glue_1.4.2                    evaluate_0.14                
## [69] BiocManager_1.30.10           httpuv_1.5.4                 
## [71] vctrs_0.3.4                   gtable_0.3.0                 
## [73] openssl_1.4.2                 purrr_0.3.4                  
## [75] assertthat_0.2.1              ggplot2_3.3.2                
## [77] xfun_0.16                     mime_0.9                     
## [79] xtable_1.8-4                  later_1.1.0.1                
## [81] survival_3.2-3                tibble_3.0.3                 
## [83] GenomicAlignments_1.24.0      memoise_1.1.0                
## [85] interactiveDisplayBase_1.26.3 ellipsis_0.3.1</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gene-symbol-mapping.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gene-sets.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-GenomicCoordinates.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to Bioconductor Annotation Resources.pdf", "Introduction to Bioconductor Annotation Resources.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
