<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Gene sets | Bioconductor Annotation Resources</title>
  <meta name="description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Gene sets | Bioconductor Annotation Resources" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Gene sets | Bioconductor Annotation Resources" />
  
  <meta name="twitter:description" content="This is a brief introduction to Bioconductor annotation resources, prepared for an R-Ladies session in Tunisia, September 30, 2020." />
  

<meta name="author" content="Martin Morgan" />


<meta name="date" content="2020-09-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="discovering-genomic-coordinates.html"/>
<link rel="next" href="advanced-working-with-sql-databases.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor Annotation Resources</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#plan"><i class="fa fa-check"></i><b>1.1</b> Plan</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#bioconductor"><i class="fa fa-check"></i><b>1.2</b> <em>Bioconductor</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html"><i class="fa fa-check"></i><b>2</b> Bulk RNA Sequencing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#rna-seq-differential-expression-data-in-r-bioconductor"><i class="fa fa-check"></i><b>2.1</b> RNA-seq differential expression data in <em>R</em> / <em>Bioconductor</em></a></li>
<li class="chapter" data-level="2.2" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#a-quick-analysis"><i class="fa fa-check"></i><b>2.2</b> A quick analysis</a></li>
<li class="chapter" data-level="2.3" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>2.3</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html"><i class="fa fa-check"></i><b>3</b> Gene symbol mapping</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>3.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="3.2" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#finding-annotation-resources"><i class="fa fa-check"></i><b>3.2</b> Finding annotation resources</a></li>
<li class="chapter" data-level="3.3" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#mapping-between-identiifiers"><i class="fa fa-check"></i><b>3.3</b> Mapping between identiifiers</a></li>
<li class="chapter" data-level="3.4" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>3.4</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html"><i class="fa fa-check"></i><b>4</b> Discovering genomic coordinates</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>4.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="4.2" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#genomic-ranges"><i class="fa fa-check"></i><b>4.2</b> Genomic ranges</a></li>
<li class="chapter" data-level="4.3" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#mapping-from-identifiers-to-ranges"><i class="fa fa-check"></i><b>4.3</b> Mapping from identifiers to ranges</a></li>
<li class="chapter" data-level="4.4" data-path="discovering-genomic-coordinates.html"><a href="discovering-genomic-coordinates.html#flexible-queries-on-ensdb-objects"><i class="fa fa-check"></i><b>4.4</b> Flexible queries on EnsDb objects</a></li>
<li class="chapter" data-level="4.5" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>4.5</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gene-sets.html"><a href="gene-sets.html"><i class="fa fa-check"></i><b>5</b> Gene sets</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>5.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="5.2" data-path="gene-sets.html"><a href="gene-sets.html#mapping-from-entrez-gene-identiifiers-to-go-identifiers"><i class="fa fa-check"></i><b>5.2</b> Mapping from (entrez) gene identiifiers to GO identifiers</a></li>
<li class="chapter" data-level="5.3" data-path="gene-sets.html"><a href="gene-sets.html#using-go-gene-ontology-annotations-to-group-genes-by-biological-process"><i class="fa fa-check"></i><b>5.3</b> Using GO (gene ontology) annotations to group genes by biological process</a></li>
<li class="chapter" data-level="5.4" data-path="gene-sets.html"><a href="gene-sets.html#more-flexible-gene-set-annotation-using-biocset"><i class="fa fa-check"></i><b>5.4</b> More flexible gene set annotation using <span>BiocSet</span></a></li>
<li class="chapter" data-level="5.5" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>5.5</b> Saving and session information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html"><i class="fa fa-check"></i><b>6</b> (Advanced) Working with SQL databases</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gene-symbol-mapping.html"><a href="gene-symbol-mapping.html#what-do-we-want-to-do"><i class="fa fa-check"></i><b>6.1</b> What do we want to do?</a></li>
<li class="chapter" data-level="6.2" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html#using-dbplyr-to-access-the-ensdb.hsapiens.v75-database"><i class="fa fa-check"></i><b>6.2</b> Using <span>dbplyr</span> to access the [EnsDb.Hsapiens.v75] database</a></li>
<li class="chapter" data-level="6.3" data-path="advanced-working-with-sql-databases.html"><a href="advanced-working-with-sql-databases.html#working-with-database-tables"><i class="fa fa-check"></i><b>6.3</b> Working with database tables</a></li>
<li class="chapter" data-level="6.4" data-path="BulkRNASeq.html"><a href="BulkRNASeq.html#saving-and-session-information"><i class="fa fa-check"></i><b>6.4</b> Saving and session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioconductor Annotation Resources</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gene-sets" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Gene sets</h1>
<div id="what-do-we-want-to-do" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> What do we want to do?</h2>
<p>Return to our top table</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="gene-sets.html#cb88-1" aria-hidden="true"></a>toptbl &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;toptbl.Rda&quot;</span>)</span>
<span id="cb88-2"><a href="gene-sets.html#cb88-2" aria-hidden="true"></a>toptbl</span></code></pre></div>
<pre><code>## # A tibble: 64,102 x 9
##    Gene  SYMBOL ENTREZID baseMean log2FoldChange  lfcSE  stat    pvalue
##    &lt;chr&gt; &lt;chr&gt;     &lt;int&gt;    &lt;dbl&gt;          &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;
##  1 ENSG… SPARC…     8404     997.           4.57 0.184   24.9 2.22e-136
##  2 ENSG… CACNB2      783     495.           3.29 0.133   24.7 7.84e-135
##  3 ENSG… DUSP1      1843    3409.           2.95 0.121   24.3 3.67e-130
##  4 ENSG… SAMHD1    25939   12703.           3.77 0.155   24.2 9.58e-130
##  5 ENSG… MAOA       4128    2342.           3.35 0.142   23.7 1.10e-123
##  6 ENSG… GPX3       2878   12286.           3.73 0.166   22.5 4.62e-112
##  7 ENSG… STEAP2   261729    3009.           1.98 0.0900  22.0 5.77e-107
##  8 ENSG… NEXN      91624    5393.           2.04 0.0942  21.6 1.32e-103
##  9 ENSG… MT2A       4502    3656.           2.21 0.106   20.9 2.90e- 97
## 10 ENSG… ADAMT…     9510   30315.           2.35 0.116   20.3 3.26e- 91
## # … with 64,092 more rows, and 1 more variable: padj &lt;dbl&gt;</code></pre>
<p>One small correction to the top table is that the ENTREZID should be a
character() vector</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="gene-sets.html#cb90-1" aria-hidden="true"></a>toptbl &lt;-<span class="st"> </span></span>
<span id="cb90-2"><a href="gene-sets.html#cb90-2" aria-hidden="true"></a><span class="st">  </span>toptbl <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb90-3"><a href="gene-sets.html#cb90-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">ENTREZID =</span> <span class="kw">as.character</span>(ENTREZID))</span></code></pre></div>
<p>Biologically, genes always function in some kind of relation with other genes
and the biology of organsims. For instance, genes
are expressed in particularly cellular compartments, or participate in
particular biological processes. We would like to use established <em>ontologies</em>
that describe these relationships.</p>
</div>
<div id="mapping-from-entrez-gene-identiifiers-to-go-identifiers" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Mapping from (entrez) gene identiifiers to GO identifiers</h2>
<p>The <a href="https://bioconductor.org/packages/org.Hs.eg.db">org.Hs.eg.db</a> packages provides a way to map from (Entrez) gene identifiers
to GO terms.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="gene-sets.html#cb91-1" aria-hidden="true"></a><span class="kw">library</span>(org.Hs.eg.db)</span></code></pre></div>
<p>Note that the keytypes of <code>org.Hs.eg.db</code> contain (Entrez) gene IDs, and the
columns contain GO ids</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="gene-sets.html#cb92-1" aria-hidden="true"></a><span class="kw">keytypes</span>(org.Hs.eg.db)</span></code></pre></div>
<pre><code>##  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot;  &quot;ENSEMBLTRANS&quot;
##  [6] &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;     &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;    
## [11] &quot;GO&quot;           &quot;GOALL&quot;        &quot;IPI&quot;          &quot;MAP&quot;          &quot;OMIM&quot;        
## [16] &quot;ONTOLOGY&quot;     &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;         &quot;PMID&quot;        
## [21] &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;       &quot;UNIGENE&quot;     
## [26] &quot;UNIPROT&quot;</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="gene-sets.html#cb94-1" aria-hidden="true"></a><span class="kw">columns</span>(org.Hs.eg.db)</span></code></pre></div>
<pre><code>##  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot;  &quot;ENSEMBLTRANS&quot;
##  [6] &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;     &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;    
## [11] &quot;GO&quot;           &quot;GOALL&quot;        &quot;IPI&quot;          &quot;MAP&quot;          &quot;OMIM&quot;        
## [16] &quot;ONTOLOGY&quot;     &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;         &quot;PMID&quot;        
## [21] &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;       &quot;UNIGENE&quot;     
## [26] &quot;UNIPROT&quot;</code></pre>
<p>Genes can belong to more than on GO category (e.g., participate in more than
one biological pathway). Use <code>AnnotationDbi::select()</code> to map between ENTREZID
and GO id; we’ll represent the result as a tibble for convenience</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="gene-sets.html#cb96-1" aria-hidden="true"></a>eggo &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db, toptbl<span class="op">$</span>ENTREZID, <span class="st">&quot;GO&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb96-2"><a href="gene-sets.html#cb96-2" aria-hidden="true"></a><span class="st">  </span><span class="kw">as_tibble</span>()</span>
<span id="cb96-3"><a href="gene-sets.html#cb96-3" aria-hidden="true"></a>eggo</span></code></pre></div>
<pre><code>## # A tibble: 364,454 x 4
##    ENTREZID GO         EVIDENCE ONTOLOGY
##    &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;   
##  1 8404     GO:0005509 IBA      MF      
##  2 8404     GO:0005515 IPI      MF      
##  3 8404     GO:0005518 IBA      MF      
##  4 8404     GO:0005576 HDA      CC      
##  5 8404     GO:0005615 HDA      CC      
##  6 8404     GO:0005615 IBA      CC      
##  7 8404     GO:0005788 TAS      CC      
##  8 8404     GO:0007165 IEA      BP      
##  9 8404     GO:0043687 TAS      BP      
## 10 8404     GO:0044267 TAS      BP      
## # … with 364,444 more rows</code></pre>
<p>Let’s focus on the ‘biological process’ (BP) ontology, and not worry about
‘EVIDENCE’ (the type of evidence, e.g., experimental verification,
computational inference, …) that supports the classification of a particular
ENTREZ identifier to a particular GO identifier.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="gene-sets.html#cb98-1" aria-hidden="true"></a>eggo &lt;-</span>
<span id="cb98-2"><a href="gene-sets.html#cb98-2" aria-hidden="true"></a><span class="st">  </span>eggo <span class="op">%&gt;%</span></span>
<span id="cb98-3"><a href="gene-sets.html#cb98-3" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">GOID =</span> <span class="st">&quot;GO&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb98-4"><a href="gene-sets.html#cb98-4" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>EVIDENCE) <span class="op">%&gt;%</span></span>
<span id="cb98-5"><a href="gene-sets.html#cb98-5" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(ONTOLOGY <span class="op">%in%</span><span class="st"> &quot;BP&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb98-6"><a href="gene-sets.html#cb98-6" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">distinct</span>()</span>
<span id="cb98-7"><a href="gene-sets.html#cb98-7" aria-hidden="true"></a>eggo</span></code></pre></div>
<pre><code>## # A tibble: 132,810 x 3
##    ENTREZID GOID       ONTOLOGY
##    &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt;   
##  1 8404     GO:0007165 BP      
##  2 8404     GO:0043687 BP      
##  3 8404     GO:0044267 BP      
##  4 8404     GO:0099560 BP      
##  5 783      GO:0007268 BP      
##  6 783      GO:0007528 BP      
##  7 783      GO:0007601 BP      
##  8 783      GO:0051928 BP      
##  9 783      GO:0061337 BP      
## 10 783      GO:0070509 BP      
## # … with 132,800 more rows</code></pre>
</div>
<div id="using-go-gene-ontology-annotations-to-group-genes-by-biological-process" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Using GO (gene ontology) annotations to group genes by biological process</h2>
<p>The <a href="https://bioconductor.org/packages/GO.db">GO.db</a> package provides information on GO identifiers.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="gene-sets.html#cb100-1" aria-hidden="true"></a><span class="kw">library</span>(GO.db)</span>
<span id="cb100-2"><a href="gene-sets.html#cb100-2" aria-hidden="true"></a><span class="kw">columns</span>(GO.db)</span></code></pre></div>
<pre><code>## [1] &quot;DEFINITION&quot; &quot;GOID&quot;       &quot;ONTOLOGY&quot;   &quot;TERM&quot;</code></pre>
<p>Use <code>AnnotationDbi::select()</code> to explore information about a particular GO id</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="gene-sets.html#cb102-1" aria-hidden="true"></a>goid &lt;-<span class="st"> &quot;GO:0007165&quot;</span></span>
<span id="cb102-2"><a href="gene-sets.html#cb102-2" aria-hidden="true"></a>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(GO.db, goid, <span class="st">&quot;TERM&quot;</span>, <span class="st">&quot;GOID&quot;</span>)</span></code></pre></div>
<pre><code>##         GOID                TERM
## 1 GO:0007165 signal transduction</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="gene-sets.html#cb104-1" aria-hidden="true"></a>def =<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(GO.db, goid, <span class="st">&quot;DEFINITION&quot;</span>, <span class="st">&quot;GOID&quot;</span>)</span>
<span id="cb104-2"><a href="gene-sets.html#cb104-2" aria-hidden="true"></a><span class="kw">cat</span>(<span class="kw">strwrap</span>(def<span class="op">$</span>DEFINITION), <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## The cellular process in which a signal is conveyed to trigger a change
## in the activity or state of a cell. Signal transduction begins with
## reception of a signal (e.g. a ligand binding to a receptor or receptor
## activation by a stimulus such as light), or for signal transduction in
## the absence of ligand, signal-withdrawal or the activity of a
## constitutively active receptor. Signal transduction ends with
## regulation of a downstream cellular process, e.g. regulation of
## transcription or regulation of a metabolic process. Signal transduction
## covers signaling from receptors located on the surface of the cell and
## signaling via molecules located within the cell. For signaling between
## cells, signal transduction is restricted to events at and within the
## receiving cell.</code></pre>
<p>Add terms and definitions to all rows of <code>eggo</code></p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="gene-sets.html#cb106-1" aria-hidden="true"></a>goinfo &lt;-<span class="st"> </span></span>
<span id="cb106-2"><a href="gene-sets.html#cb106-2" aria-hidden="true"></a><span class="st">  </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(</span>
<span id="cb106-3"><a href="gene-sets.html#cb106-3" aria-hidden="true"></a>    GO.db, eggo<span class="op">$</span>GOID, <span class="kw">c</span>(<span class="st">&quot;TERM&quot;</span>, <span class="st">&quot;DEFINITION&quot;</span>), <span class="st">&quot;GOID&quot;</span></span>
<span id="cb106-4"><a href="gene-sets.html#cb106-4" aria-hidden="true"></a>  ) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb106-5"><a href="gene-sets.html#cb106-5" aria-hidden="true"></a><span class="st">  </span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span></span>
<span id="cb106-6"><a href="gene-sets.html#cb106-6" aria-hidden="true"></a><span class="st">  </span><span class="kw">distinct</span>()</span>
<span id="cb106-7"><a href="gene-sets.html#cb106-7" aria-hidden="true"></a>goinfo</span></code></pre></div>
<pre><code>## # A tibble: 12,130 x 3
##    GOID     TERM                     DEFINITION                                 
##    &lt;chr&gt;    &lt;chr&gt;                    &lt;chr&gt;                                      
##  1 GO:0007… signal transduction      The cellular process in which a signal is …
##  2 GO:0043… post-translational prot… The process of covalently altering one or …
##  3 GO:0044… cellular protein metabo… The chemical reactions and pathways involv…
##  4 GO:0099… synaptic membrane adhes… The attachment of presynaptic membrane to …
##  5 GO:0007… chemical synaptic trans… The vesicular release of classical neurotr…
##  6 GO:0007… neuromuscular junction … A process that is carried out at the cellu…
##  7 GO:0007… visual perception        The series of events required for an organ…
##  8 GO:0051… positive regulation of … Any process that activates or increases th…
##  9 GO:0061… cardiac conduction       Transfer of an organized electrical impuls…
## 10 GO:0070… calcium ion import       The directed movement of calcium ions into…
## # … with 12,120 more rows</code></pre>
<p>Use <code>dplyr::left_join()</code> to join all rows of <code>eggo</code> to their corresponding
term and definition…</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="gene-sets.html#cb108-1" aria-hidden="true"></a>eggoinfo &lt;-<span class="st"> </span><span class="kw">left_join</span>(eggo, goinfo, <span class="dt">by =</span> <span class="st">&quot;GOID&quot;</span>)</span>
<span id="cb108-2"><a href="gene-sets.html#cb108-2" aria-hidden="true"></a>eggoinfo</span></code></pre></div>
<pre><code>## # A tibble: 132,810 x 5
##    ENTREZID GOID    ONTOLOGY TERM               DEFINITION                      
##    &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;              &lt;chr&gt;                           
##  1 8404     GO:000… BP       signal transducti… The cellular process in which a…
##  2 8404     GO:004… BP       post-translationa… The process of covalently alter…
##  3 8404     GO:004… BP       cellular protein … The chemical reactions and path…
##  4 8404     GO:009… BP       synaptic membrane… The attachment of presynaptic m…
##  5 783      GO:000… BP       chemical synaptic… The vesicular release of classi…
##  6 783      GO:000… BP       neuromuscular jun… A process that is carried out a…
##  7 783      GO:000… BP       visual perception  The series of events required f…
##  8 783      GO:005… BP       positive regulati… Any process that activates or i…
##  9 783      GO:006… BP       cardiac conduction Transfer of an organized electr…
## 10 783      GO:007… BP       calcium ion import The directed movement of calciu…
## # … with 132,800 more rows</code></pre>
<p>…and even add the top table gene expression information</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="gene-sets.html#cb110-1" aria-hidden="true"></a><span class="kw">left_join</span>(eggoinfo, toptbl, <span class="dt">by =</span> <span class="st">&quot;ENTREZID&quot;</span>)</span></code></pre></div>
<pre><code>## # A tibble: 155,800 x 13
##    ENTREZID GOID  ONTOLOGY TERM  DEFINITION Gene  SYMBOL baseMean log2FoldChange
##    &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt;          &lt;dbl&gt;
##  1 8404     GO:0… BP       sign… The cellu… ENSG… SPARC…     997.           4.57
##  2 8404     GO:0… BP       post… The proce… ENSG… SPARC…     997.           4.57
##  3 8404     GO:0… BP       cell… The chemi… ENSG… SPARC…     997.           4.57
##  4 8404     GO:0… BP       syna… The attac… ENSG… SPARC…     997.           4.57
##  5 783      GO:0… BP       chem… The vesic… ENSG… CACNB2     495.           3.29
##  6 783      GO:0… BP       neur… A process… ENSG… CACNB2     495.           3.29
##  7 783      GO:0… BP       visu… The serie… ENSG… CACNB2     495.           3.29
##  8 783      GO:0… BP       posi… Any proce… ENSG… CACNB2     495.           3.29
##  9 783      GO:0… BP       card… Transfer … ENSG… CACNB2     495.           3.29
## 10 783      GO:0… BP       calc… The direc… ENSG… CACNB2     495.           3.29
## # … with 155,790 more rows, and 4 more variables: lfcSE &lt;dbl&gt;, stat &lt;dbl&gt;,
## #   pvalue &lt;dbl&gt;, padj &lt;dbl&gt;</code></pre>
</div>
<div id="more-flexible-gene-set-annotation-using-biocset" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> More flexible gene set annotation using <a href="https://bioconductor.org/packages/BiocSet">BiocSet</a></h2>
<p>Gene sets can be thought of as a three related ‘tables’: genes, sets, and the
gene/set relationship. <a href="https://bioconductor.org/packages/BiocSet">BiocSet</a> provides a way to manage these tables.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="gene-sets.html#cb112-1" aria-hidden="true"></a><span class="kw">library</span>(BiocSet)</span></code></pre></div>
<p>Start by identifying the unique element / set combinations</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="gene-sets.html#cb113-1" aria-hidden="true"></a>elementsets &lt;-</span>
<span id="cb113-2"><a href="gene-sets.html#cb113-2" aria-hidden="true"></a><span class="st">  </span>eggo <span class="op">%&gt;%</span></span>
<span id="cb113-3"><a href="gene-sets.html#cb113-3" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="dt">element =</span> <span class="st">&quot;ENTREZID&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;GOID&quot;</span>)</span></code></pre></div>
<p>Then create a BiocSet from this relationship</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="gene-sets.html#cb114-1" aria-hidden="true"></a>es &lt;-<span class="st"> </span><span class="kw">BiocSet_from_elementset</span>(elementsets)</span>
<span id="cb114-2"><a href="gene-sets.html#cb114-2" aria-hidden="true"></a>es</span></code></pre></div>
<pre><code>## class: BiocSet
## 
## es_element():
## # A tibble: 17,246 x 1
##   element
##   &lt;chr&gt;  
## 1 1890   
## 2 4205   
## 3 291    
## # … with 17,243 more rows
## 
## es_set():
## # A tibble: 12,130 x 1
##   set       
##   &lt;chr&gt;     
## 1 GO:0000002
## 2 GO:0000003
## 3 GO:0000012
## # … with 12,127 more rows
## 
## es_elementset() &lt;active&gt;:
## # A tibble: 132,810 x 2
##   element set       
##   &lt;chr&gt;   &lt;chr&gt;     
## 1 1890    GO:0000002
## 2 4205    GO:0000002
## 3 291     GO:0000002
## # … with 132,807 more rows</code></pre>
<p>Use <code>left_join_*()</code> to join annotations to the elements and sets, first removing
the duplicate information in <code>eggoinfo</code> cause by including the ‘GOID’ column</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="gene-sets.html#cb116-1" aria-hidden="true"></a>eggoinfo1 &lt;-</span>
<span id="cb116-2"><a href="gene-sets.html#cb116-2" aria-hidden="true"></a><span class="st">  </span>eggoinfo <span class="op">%&gt;%</span></span>
<span id="cb116-3"><a href="gene-sets.html#cb116-3" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>ENTREZID) <span class="op">%&gt;%</span></span>
<span id="cb116-4"><a href="gene-sets.html#cb116-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">distinct</span>()</span>
<span id="cb116-5"><a href="gene-sets.html#cb116-5" aria-hidden="true"></a>es &lt;-</span>
<span id="cb116-6"><a href="gene-sets.html#cb116-6" aria-hidden="true"></a><span class="st">  </span>es <span class="op">%&gt;%</span></span>
<span id="cb116-7"><a href="gene-sets.html#cb116-7" aria-hidden="true"></a><span class="st">  </span><span class="kw">left_join_element</span>(toptbl, <span class="dt">by =</span> <span class="kw">c</span>(<span class="dt">element =</span> <span class="st">&quot;ENTREZID&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb116-8"><a href="gene-sets.html#cb116-8" aria-hidden="true"></a><span class="st">  </span><span class="kw">left_join_set</span>(eggoinfo1, <span class="dt">by =</span> <span class="kw">c</span>(<span class="dt">set =</span> <span class="st">&quot;GOID&quot;</span>))</span></code></pre></div>
<p>The result is an object that coordinates annotations on genes and sets</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="gene-sets.html#cb117-1" aria-hidden="true"></a>es</span></code></pre></div>
<pre><code>## class: BiocSet
## 
## es_element():
## # A tibble: 20,003 x 9
##   element Gene     SYMBOL baseMean log2FoldChange  lfcSE  stat   pvalue     padj
##   &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;     &lt;dbl&gt;          &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 1890    ENSG000… TYMP       145.          1.23  0.199   6.18 6.58e-10  1.49e-8
## 2 4205    ENSG000… MEF2A     1566.          0.529 0.0919  5.76 8.45e- 9  1.64e-7
## 3 291     ENSG000… SLC25…     361.          0.660 0.142   4.65 3.35e- 6  4.15e-5
## # … with 20,000 more rows
## 
## es_set():
## # A tibble: 12,130 x 4
##   set      ONTOLOGY TERM               DEFINITION                               
##   &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;              &lt;chr&gt;                                    
## 1 GO:0000… BP       mitochondrial gen… The maintenance of the structure and int…
## 2 GO:0000… BP       reproduction       The production of new individuals that c…
## 3 GO:0000… BP       single strand bre… The repair of single strand breaks in DN…
## # … with 12,127 more rows
## 
## es_elementset() &lt;active&gt;:
## # A tibble: 132,810 x 2
##   element set       
##   &lt;chr&gt;   &lt;chr&gt;     
## 1 1890    GO:0000002
## 2 4205    GO:0000002
## 3 291     GO:0000002
## # … with 132,807 more rows</code></pre>
<p>The individual tables are easily manipulated</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="gene-sets.html#cb119-1" aria-hidden="true"></a><span class="kw">es_set</span>(es) <span class="op">%&gt;%</span></span>
<span id="cb119-2"><a href="gene-sets.html#cb119-2" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">startsWith</span>(TERM, <span class="st">&quot;mitochondrial&quot;</span>))</span></code></pre></div>
<pre><code>## # A tibble: 70 x 4
##    set     ONTOLOGY TERM                      DEFINITION                        
##    &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;                     &lt;chr&gt;                             
##  1 GO:000… BP       mitochondrial genome mai… The maintenance of the structure …
##  2 GO:000… BP       mitochondrial fission     The division of a mitochondrion w…
##  3 GO:000… BP       mitochondrial RNA catabo… The chemical reactions and pathwa…
##  4 GO:000… BP       mitochondrial mRNA catab… The chemical reactions and pathwa…
##  5 GO:000… BP       mitochondrial RNA metabo… The chemical reactions and pathwa…
##  6 GO:000… BP       mitochondrial RNA proces… The conversion of a primary RNA m…
##  7 GO:000… BP       mitochondrial RNA 5&#39;-end… Any process involved in forming t…
##  8 GO:000… BP       mitochondrial RNA 3&#39;-end… Any process involved in forming t…
##  9 GO:000… BP       mitochondrial electron t… The transfer of electrons from NA…
## 10 GO:000… BP       mitochondrial electron t… The transfer of electrons from su…
## # … with 60 more rows</code></pre>
<p>as are the elements and sets</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="gene-sets.html#cb121-1" aria-hidden="true"></a>goids &lt;-</span>
<span id="cb121-2"><a href="gene-sets.html#cb121-2" aria-hidden="true"></a><span class="st">  </span><span class="kw">es_set</span>(es) <span class="op">%&gt;%</span></span>
<span id="cb121-3"><a href="gene-sets.html#cb121-3" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">startsWith</span>(TERM, <span class="st">&quot;mitochondrial&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb121-4"><a href="gene-sets.html#cb121-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">pull</span>(set)</span>
<span id="cb121-5"><a href="gene-sets.html#cb121-5" aria-hidden="true"></a>mitochondrial_genes &lt;-</span>
<span id="cb121-6"><a href="gene-sets.html#cb121-6" aria-hidden="true"></a><span class="st">  </span>es <span class="op">%&gt;%</span></span>
<span id="cb121-7"><a href="gene-sets.html#cb121-7" aria-hidden="true"></a><span class="st">  </span><span class="kw">filter_elementset</span>(set <span class="op">%in%</span><span class="st"> </span>goids)</span></code></pre></div>
<p>Sets can be converted to ‘denormalized’ tibbles</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="gene-sets.html#cb122-1" aria-hidden="true"></a>mitochondrial_genes <span class="op">%&gt;%</span></span>
<span id="cb122-2"><a href="gene-sets.html#cb122-2" aria-hidden="true"></a><span class="st">  </span><span class="kw">tibble_from_elementset</span>()</span></code></pre></div>
<pre><code>## # A tibble: 764 x 13
##    element set   ONTOLOGY TERM  DEFINITION Gene  SYMBOL baseMean log2FoldChange
##    &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt;          &lt;dbl&gt;
##  1 1890    GO:0… BP       mito… The maint… ENSG… TYMP      145.           1.23 
##  2 4205    GO:0… BP       mito… The maint… ENSG… MEF2A    1566.           0.529
##  3 291     GO:0… BP       mito… The maint… ENSG… SLC25…    361.           0.660
##  4 10000   GO:0… BP       mito… The maint… ENSG… AKT3     1195.           0.327
##  5 9361    GO:0… BP       mito… The maint… ENSG… LONP1    1215.          -0.326
##  6 55186   GO:0… BP       mito… The maint… ENSG… SLC25…   2065.           0.145
##  7 92667   GO:0… BP       mito… The maint… ENSG… MGME1     290.           0.202
##  8 4976    GO:0… BP       mito… The maint… ENSG… OPA1     1309.           0.123
##  9 4976    GO:0… BP       mito… The maint… LRG_… LRG_3…      0           NA    
## 10 84275   GO:0… BP       mito… The maint… ENSG… SLC25…     65.9          0.305
## # … with 754 more rows, and 4 more variables: lfcSE &lt;dbl&gt;, stat &lt;dbl&gt;,
## #   pvalue &lt;dbl&gt;, padj &lt;dbl&gt;</code></pre>
</div>
<div id="saving-and-session-information" class="section level2" number="5.5">
<h2><span class="header-section-number">5.5</span> Saving and session information</h2>
<p>As usual, we close the chapter by saving the primary result</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="gene-sets.html#cb124-1" aria-hidden="true"></a><span class="kw">saveRDS</span>(es, <span class="st">&quot;GO_elementset.Rda&quot;</span>)</span></code></pre></div>
<p>and recording session information.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="gene-sets.html#cb125-1" aria-hidden="true"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.2 Patched (2020-09-15 r79213)
## Platform: x86_64-apple-darwin17.7.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS:   /Users/ma38727/bin/R-4-0-branch/lib/libRblas.dylib
## LAPACK: /Users/ma38727/bin/R-4-0-branch/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] BiocSet_1.2.2               AnnotationHub_2.20.2       
##  [3] BiocFileCache_1.12.1        dbplyr_1.4.4               
##  [5] DESeq2_1.28.1               airway_1.8.0               
##  [7] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
##  [9] matrixStats_0.56.0          org.Hs.eg.db_3.11.4        
## [11] GO.db_3.11.4                dplyr_1.0.2                
## [13] EnsDb.Hsapiens.v75_2.99.0   ensembldb_2.12.1           
## [15] AnnotationFilter_1.12.0     GenomicFeatures_1.40.1     
## [17] AnnotationDbi_1.50.3        Biobase_2.48.0             
## [19] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
## [21] IRanges_2.22.2              S4Vectors_0.26.1           
## [23] BiocGenerics_0.34.0        
## 
## loaded via a namespace (and not attached):
##  [1] ProtGenerics_1.20.0           bitops_1.0-6                 
##  [3] bit64_4.0.5                   RColorBrewer_1.1-2           
##  [5] progress_1.2.2                httr_1.4.2                   
##  [7] tools_4.0.2                   utf8_1.1.4                   
##  [9] R6_2.4.1                      DBI_1.1.0                    
## [11] lazyeval_0.2.2                colorspace_1.4-1             
## [13] tidyselect_1.1.0              prettyunits_1.1.1            
## [15] bit_4.0.4                     curl_4.3                     
## [17] compiler_4.0.2                cli_2.0.2                    
## [19] rtracklayer_1.48.0            bookdown_0.20                
## [21] scales_1.1.1                  genefilter_1.70.0            
## [23] askpass_1.1                   rappdirs_0.3.1               
## [25] stringr_1.4.0                 digest_0.6.25                
## [27] Rsamtools_2.4.0               rmarkdown_2.3                
## [29] XVector_0.28.0                pkgconfig_2.0.3              
## [31] htmltools_0.5.0               fastmap_1.0.1                
## [33] rlang_0.4.7                   rstudioapi_0.11              
## [35] RSQLite_2.2.0                 shiny_1.5.0                  
## [37] generics_0.0.2                BiocParallel_1.22.0          
## [39] RCurl_1.98-1.2                magrittr_1.5                 
## [41] GenomeInfoDbData_1.2.3        Matrix_1.2-18                
## [43] fansi_0.4.1                   Rcpp_1.0.5                   
## [45] munsell_0.5.0                 lifecycle_0.2.0              
## [47] stringi_1.4.6                 yaml_2.2.1                   
## [49] zlibbioc_1.34.0               plyr_1.8.6                   
## [51] grid_4.0.2                    blob_1.2.1                   
## [53] promises_1.1.1                crayon_1.3.4                 
## [55] lattice_0.20-41               Biostrings_2.56.0            
## [57] splines_4.0.2                 annotate_1.66.0              
## [59] KEGGREST_1.28.0               hms_0.5.3                    
## [61] locfit_1.5-9.4                knitr_1.29                   
## [63] pillar_1.4.6                  codetools_0.2-16             
## [65] geneplotter_1.66.0            biomaRt_2.44.1               
## [67] XML_3.99-0.5                  glue_1.4.2                   
## [69] BiocVersion_3.11.1            evaluate_0.14                
## [71] BiocManager_1.30.10           png_0.1-7                    
## [73] httpuv_1.5.4                  vctrs_0.3.4                  
## [75] gtable_0.3.0                  openssl_1.4.2                
## [77] purrr_0.3.4                   assertthat_0.2.1             
## [79] ggplot2_3.3.2                 xfun_0.16                    
## [81] mime_0.9                      xtable_1.8-4                 
## [83] later_1.1.0.1                 survival_3.2-3               
## [85] tibble_3.0.3                  GenomicAlignments_1.24.0     
## [87] memoise_1.1.0                 ellipsis_0.3.1               
## [89] interactiveDisplayBase_1.26.3</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="discovering-genomic-coordinates.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="advanced-working-with-sql-databases.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-GeneSets.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to Bioconductor Annotation Resources.pdf", "Introduction to Bioconductor Annotation Resources.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
